src/messages/Message.hpp

```diff
@@ -63,6 +63,7 @@ struct Message : boost::noncopyable {
     QString timeoutUser;
     QString channelName;
     QColor usernameColor;
+    long receivedTime;
```

Comment:
2 things here:
- it will also hold backwards dates for historical messages, so calling it something like sentTime or produceTime would fit better
- not a fan of long, you could use QDateTime, then pass .time() for TimestampElement ctor and and use it in your sorting just as fine.

Reply:
Thanks for the feedback. I believe I've resolved both of those concerns in my latest commits.